src/lib/controllers/deploy-adapters/index.ts

```diff
@@ -0,0 +1,51 @@
+import * as Promise from 'bluebird';
+import { ComposeDeployment } from './compose';
+import { KubernetesDeployment } from './kubernetes';
+
+export const AvailableDeployAdapters = [
+	'balena',
+	'compose',
+	'docker-compose',
+	'kubernetes',
+];
+
+export class UnknownDeploymentTargetError extends Error {
+	constructor(target: string) {
+		super(`Unknown target: ${target}`);
+	}
+}
+
+export function getDeployAdapter(
+	target: string,
+	config: DeployConfig,
+): Deployable {
+	switch (target.toLowerCase().trim()) {
+		case 'balena':
```

Comment:
We should probably need to handle `balena` via the balena deploy adapter here or remove it. (I see this was probably lost in the review process).

Reply:
np, I will drop it down to the unsupported option for now.